jetty-fcgi/fcgi-server/src/main/java/org/eclipse/jetty/fcgi/server/ServerFCGIConnection.java

```diff
@@ -117,51 +118,109 @@ else if (read == 0)
         {
             if (LOG.isDebugEnabled())
                 LOG.debug("Unable to fill endpoint", x);
-            bufferPool.release(buffer);
+            networkBuffer.clear();
+            releaseInputBuffer();
             // TODO: fail and close ?
         }
     }
 
+    /**
+     * This is just a "consume" method, so it must not call
+     * fillInterested(), but just consume what's in the network
+     * for the current request.
+     */
+    void parseAndFill()
+    {
+        if (LOG.isDebugEnabled())
+            LOG.debug("parseAndFill {}", this);
+        while (channel != null)
```

Comment:
I think it would be worthwhile to add a comment stating that this is the "committee loop", that it matches what HTTP/1 is doing and that it shouldn't be touched lightly despite the fact that it's counter-intuitive.

Reply:
Done.